clientv3/auth.go

```diff
@@ -61,7 +61,7 @@ type Auth interface {
 	AuthDisable(ctx context.Context) (*AuthDisableResponse, error)
 
 	// UserAdd adds a new user to an etcd cluster.
-	UserAdd(ctx context.Context, name string, password string) (*AuthUserAddResponse, error)
+	UserAdd(ctx context.Context, name string, password string, noPassword bool) (*AuthUserAddResponse, error)
```

Comment:
This is a source backward incompatible change in the client.  Is this acceptable?  The only alternative I can think of would be to introduce an additional `UserAddNoPassword` function, or similar.

cc @gyuho for compatibility expectations of changes to client

Reply:
@jpbetz Oh good catch.

`UserAddNoPassword` sounds better for compatibilities.

/cc @mitake

Reply:
One other alternative would be to introduce a `UserAddWithOptons(..., opts... UserAddOption)` function as well as a `WithNoPassword` `UserAddOption`.  This would future proof the API against additions of more UserAdd parameters in the future.  I'm not sure if that's justified here or not.

Reply:
Thanks for pointing out. I'll introduce `UserAddWithOptons`. It would be good for expanding in the future as @jpbetz says.